test/ZToken.test.ts

```diff
@@ -115,6 +116,75 @@ describe("ZToken Test", () => {
       ).withArgs(inflationRatesInvalid);
     });
 
+    it("Should return 0 each year if INLFATION RATES passed as zeros", async () => {
+      const inflationRatesInvalid = [0n, 0n, 0n, 0n, 0n, 0n, 0n, 0n, 0n, 0n, 0n, 0n];
+
+      const token = await ZTokenFactory.deploy(
+        tokenName,
+        tokenSymbol,
+        admin.address,
+        admin.address,
+        beneficiary.address,
+        INITIAL_SUPPLY_DEFAULT,
+        inflationRatesInvalid,
+        FINAL_INFLATION_RATE_DEFAULT,
+      );
+
+      let time = deployTime;
+
+      // From 1 `year` to `inflationRatesInvalid.length` - 1 cause first rate is 0n,
+      // and last rate should be 1.5%
+      for (let year = 1; year < inflationRatesInvalid.length; year++) {
+        time += 31536000n;
+
+        expect(
+          await token.calculateMintableTokens(time)
+        ).to.be.equal(
+          0n
+        );
+      }
+    });
+
+    it("Should return 0 tokens in 13 year if FINAL INLFATION RATE is passed as 0n", async () => {
+      const finalRate = 0n;
+
+      const token = await ZTokenFactory.deploy(
+        tokenName,
+        tokenSymbol,
+        admin.address,
+        admin.address,
+        beneficiary.address,
+        INITIAL_SUPPLY_DEFAULT,
+        INFLATION_RATES_DEFAULT,
+        finalRate,
+      );
+
+      const time = deployTime + 31536000n * 13n;
+
+      const mintAmount = await token.calculateMintableTokens(time);
+
+      let sum = 0n;
+      for (let year = 1; year <= INFLATION_RATES_DEFAULT.length; year++) {
+        sum += await token.tokensPerYear(year);
+      }
+
+      const rate = await token.currentInflationRate(
+        (time - deployTime) / 31536000n
+      );
+
+      expect(
+        mintAmount
+      ).to.be.equal(
+        sum
+      );
```

Comment:
you are testing returns from the same functions. this is not a test.
test values from different places against each other, not the same place.
test against TS calculated values.